src/main/java/com/liferay/faces/util/i18n/I18nBundleBase.java

```diff
@@ -183,24 +170,14 @@ public I18n getWrapped() {
 	}
 
 	/**
-	 * Returns the initial cache capacity of the {@link I18nBundleBase}. The default implementation returns the provided
-	 * default value. This method is called from the constructor of I18nBundleBase, so this method must not cause side
-	 * effects and cannot rely on I18nBundleBase being fully initialized. The default return value is null.
-	 *
-	 * @param  defaultInitialCacheCapacity  The default initial capacity of the cache.
-	 */
-	protected Integer getInitialCacheCapacity(int defaultInitialCacheCapacity) {
-		return defaultInitialCacheCapacity;
-	}
-
-	/**
-	 * Returns the max cache capacity of the {@link I18nBundleBase} or null if the cache should not have a maximum size.
-	 * This method is called from the constructor of I18nBundleBase, so this method must not cause side effects and
-	 * cannot rely on I18nBundleBase being fully initialized. The default return value is null.
+	 * Returns a new message cache to be used by {@link I18nBundleBase}. The default implementation returns a Cache
+	 * instance obtained from {@link CacheFactory#getConcurrentCacheInstance(javax.faces.context.ExternalContext)}. This
+	 * method is called from the constructor of I18nBundleBase, so this method must not cause side effects and should
+	 * not expect I18nBundleBase (or its subclass) to be fully initialized.
 	 *
-	 * @param  externalContext  The external context associated with the current faces context.
+	 * @param  externalContext  The external context associated with the current {@link FacesContext}.
 	 */
-	protected Integer getMaxCacheCapacity(ExternalContext externalContext) {
-		return null;
+	protected Cache<String, String> newConcurrentMessageCache(ExternalContext externalContext) {
```

Comment:
@stiemannkj1 I18nBundleBase.java is API, so adding and removing `protected` methods will cause a breaking API change. Can this be avoided? (Just in case we are not the only ones trying to extend this).